_posts/2020-02-22-gsoc2020.md

```diff
@@ -0,0 +1,113 @@
+| 2 | [Command Line Interface (CLI)]({{ site.github.wiki }}/Project-Ideas-2020#project-2-command-line-interface-cli) | easy | create a new package with MDAnalysis-based command line tools | Python, bash, Python packages, CI, MD | @PicoCentauri @orbeckst @fiona-naughton
+| 3 | [Interoperability with RDKit]({{ site.github.wiki }}/Project-Ideas-2020#project-3-interoperability-with-rdkit) | medium | add capability to MDAnalysis to use the RDKit API to convert data structures between MDAnalysis and RDKit | MDAnalysis, RDKit, Python, C++ (?) | @richardjgowers @IAlibay
+| 4 | [Improved atom selections]({{ site.github.wiki }}/Project-Ideas-2020#project-4-improved-atom-selections) | medium | replace the selection parsing code with a more flexible parser | Python | @orbeckst, @IAlibay, @fiona-naughton
+| 5 | [Serialize Universes for parallel computation]({{ site.github.wiki }}/Project-Ideas-2020#project-5-serialize-universes-for-parallel-computation) | medium/challenging | make the central `Universe` data structure serializable with Pickle to enable simple parallelization with Dask or MPI | Python, MDAnalysis I/O, task-based parallelization | @richardjgowers, @orbeckst, @IAlibay, @fiona-naughton 
+| 6 | [Implement TNG support]({{ site.github.wiki }}/Project-Ideas-2020#project-6-implement-tng-support) | challenging | write Cython bindings to the TNG library and write classes to bring this information into MDAnalysis | Cython, C++ | @richardjgowers
+
+# Information for Students
+
+You must meet our [own requirements] if you want to be a student with MDAnalysis
+this year (read all the docs behind these links!). You must also meet the
+[eligibility criteria].
+
+The MDAnalysis community values diversity and is committed to
+providing a productive, harassment-free to every member. Our [Code of
+Conduct] explains the values that we as a community uphold. *Every
+member (and every GSoC student) agrees to follow the [Code of
+Conduct]*.
+
+
+As a start to get familiar with MDAnalysis and open source development you
+should follow these steps:
+
+## Complete the Quick Start Guide
+
+We have a [Quick Start Guide] explaining the basics of MDAnalysis. You
+should go through it at least once to understand how MDAnalysis is
+used. Continue reading the [User Guide] to learn more.
+
+## Introduce yourself to us
+
+Introduce yourself on the [mailing list]. Tell us what you plan to work
+on during the summer or what you have already done with MDAnalysis.
+
+## Close an issue of MDAnalysis
+
+You must have *at least one commit in the development branch of
+MDAnalysis* in order to be eligible, i.e. you must demonstrate that
+you have been seriously engaged with the MDAnalysis project.
+We have a list of [easy bugs][] and suggested [GSOC Starter
```

Comment:
What's the distinction between `GSOC Starter` and `Difficulty-easy` issues? I don't know if some of the "easy bugs" are actually easy, e.g. I would expect [rewriting helanal to use AnalysisBase](https://github.com/MDAnalysis/mdanalysis/issues/2452) would require way more work than [cleaning up PSAnalysis](https://github.com/MDAnalysis/mdanalysis/issues/2463). It would be nice to have a tag for absolute beginners to Git/open-source/molecular dynamics/MDAnalysis with issues that could be solved in a couple hours, eg.:

- [Add requirements.txt and environment.yml for dependencies](https://github.com/MDAnalysis/mdanalysis/issues/2543)
- [fix PCA](https://github.com/MDAnalysis/mdanalysis/issues/2510)
- [fix AnalysisFromFunction](https://github.com/MDAnalysis/mdanalysis/issues/2503)
- [Document AlignTraj.match_atoms](https://github.com/MDAnalysis/mdanalysis/issues/2469)
- [RMSD groupselections](https://github.com/MDAnalysis/mdanalysis/issues/2429)
- [Merge Universes without coordinates](https://github.com/MDAnalysis/mdanalysis/issues/2470)
- [adding angles for SETTLES](https://github.com/MDAnalysis/mdanalysis/issues/2417)(if issue is necessary, close if not)
- [Drop TRZ force-writing and fix DL_Poly units](https://github.com/MDAnalysis/mdanalysis/issues/2393)
- [Contact analysis with PBC](https://github.com/MDAnalysis/mdanalysis/issues/2368)

And User Guide could also link to those beginner issues in "Where to Start?".

Reply:
Difficulty is an odd one - there's some things that are quick (if how long the patch will take is even the metric here) if you know the tangled history of MDA, but otherwise are a nightmare.

There's also how well scoped out the issue is. The PCA one you've is currently undiagnosed, so it could be a nightmare to track down or a single line change.  The `mda.Merge` one is the opposite, nicely scoped out and probably just a few lines to fix.

I think a starter definitely has to be easy for someone seeing the repo for the first time

Reply:
@richardjgowers it is just a single line change to an if condition, I just didn't put it in the issue because I thought I'd get around to fixing it sooner. Most of the issues in that list are mine because I do mostly know where the issue is and that testing the fix isn't too hard; editing them now.

Reply:
haha ok, well it'll be at least 2 lines with the test :p

Reply:
@lilyminium please outline the fix in the issue and mark them beginner. This way they can be used to familiarize people with our workflow.

Reply:
@lilyminium please outline the fix in the issue and mark them beginner. This way they can be used to familiarize people with our workflow.

Reply:
@kain88-de I've defined the problems more and added notes on potential fixes/test cases/pointers to examples. Let me know if I should more explicitly lay out a possible workflow. I don't have permissions to label issues so could you do that?

Reply:
done